ndk/hooks/useFeedWithEose.ts

```diff
@@ -1,39 +1,75 @@
-import { type NDKEvent, type NDKFilter } from "@nostr-dev-kit/ndk";
+import {
+  type NDKEvent,
+  type NDKFilter,
+  NDKSubscription,
+} from "@nostr-dev-kit/ndk";
 import { useNDK } from "ndk/NDKProvider";
 import { createRelaySet } from "ndk/utils";
 import { useEffect, useRef, useState } from "react";
+import { chunkArray } from "../../utils/common";
 
 export function useFeedWithEose(filter: NDKFilter, relayUrls: string[] = []) {
   const { ndk } = useNDK();
   const [feed, setFeed] = useState<NDKEvent[]>([]);
   const eventBatch = useRef<NDKEvent[] | null>([]);
+  const eoseCount = useRef(0);
+  const sortedFeed = feed.sort(
+    (a, b) => (b.created_at ?? 0) - (a.created_at ?? 0)
+  );
 
   useEffect(() => {
     if (!ndk) {
       return;
     }
 
     const relaySet = createRelaySet(relayUrls, ndk);
-    const subscription = ndk.subscribe(filter, undefined, relaySet);
-
-    subscription.on("event", (event: NDKEvent) => {
-      if (eventBatch.current === null) {
-        setFeed((prev) => [event, ...prev]);
-      } else {
-        eventBatch.current.push(event);
-      }
+
+    // many relays have a max of 256 authors
+    const maxAuthors = 256;
+
+    const filters =
+      filter.authors?.length ?? 0 > maxAuthors
+        ? chunkArray(filter.authors, maxAuthors).map((authors) => ({
+            ...filter,
+            authors,
+          }))
+        : [filter];
```

Comment:
This logic should be moved into chunkArray 
```
function chunkArray(array, max) {
  if (array.length <= max)
    return [array];
  ...
```

Reply:
Don't even need to do that. I'm regarded. Will clean up in a min.

Reply:
Actually, we do need this logic in case there's no authors in the filter.

Reply:
> function chunkArray(array, max) {
  if (array.length <= max)
    return [array];
  ...

If statement isn't necessary. The loop logic does this even when the length is less than the max.

> const filters = chunkArray(filter.authors, maxAuthors)

This would set filters to a chunked array of authors. 

The logic I have here is for when the filter doesn't have authors e.g. 

```
{ kinds: [7], "#e": ["ee6cebc9e016a48dfcc02c6f48da7f0b7902e604d1f25e2c0d8cb8939112ca64"] }
```

Reply:
Let me add a variable to make the code easier to understand.

Reply:
my logic was messed up. haha not sure why i had `maxAuthors` in the conditional.

Reply:
oooh yeah i see now, i misread the code.

Reply:
yeah my bad. the code was wrong. hahaha. yeah good to go.